lucene/core/src/java/org/apache/lucene/index/ReaderPool.java

```diff
@@ -404,7 +404,7 @@ private PendingDeletes newPendingDeletes(SegmentReader reader, SegmentCommitInfo
   private boolean noDups() {
     Set<String> seen = new HashSet<>();
     for(SegmentCommitInfo info : readerMap.keySet()) {
-      assert !seen.contains(info.info.name);
+      assert !seen.contains(info.info.name) : "seen twice: " + info.info.name ;
```

Comment:
I love seeing diffs like this one, adding a `String` message to an otherwise cryptic `assert`!  It makes me realize you must have had a hellacious debugging session!

Reply:
many fun issues in this PR to be honest. IW is tricky as hell in some places like we are incRefing files in StandardDirectoryReader but not in IW for NRT readers is mindblowing :D